polaris-react/src/components/DataTable/DataTable.tsx

```diff
@@ -219,6 +320,59 @@ class DataTableInner extends PureComponent<CombinedProps, DataTableState> {
     );
   }
 
+  private setCellRef = ({
+    ref,
+    index,
+    inStickyHeader,
+  }: {
+    ref: HTMLTableCellElement | null;
+    index: number;
+    inStickyHeader: boolean;
+  }) => {
+    if (ref == null) {
+      return;
+    }
+
+    if (inStickyHeader) {
+      this.stickyHeadings[index] = ref;
+    } else {
+      this.tableHeadings[index] = ref;
+      this.tableHeadingWidths[index] = ref.offsetWidth;
+    }
+  };
+
+  private changeHeadingFocus = () => {
+    const {tableHeadings, stickyHeadings} = this;
+
+    const stickyFocusedItemIndex = stickyHeadings.findIndex(
+      (item) => item === document.activeElement?.parentElement,
+    );
+
+    const tableFocusedItemIndex = tableHeadings.findIndex(
+      (item) => item === document.activeElement?.parentElement,
+    );
+
+    if (stickyFocusedItemIndex < 0 && tableFocusedItemIndex < 0) {
+      return null;
+    }
+
+    let button;
+
+    if (stickyFocusedItemIndex >= 0) {
+      button = tableHeadings[stickyFocusedItemIndex].querySelector('button');
+    } else if (tableFocusedItemIndex >= 0) {
+      button = stickyHeadings[tableFocusedItemIndex].querySelector('button');
+    }
+
+    if (button == null) {
+      return null;
+    }
+
+    button.style.visibility = 'visible';
+    button.focus();
+    button.removeAttribute('style');
```

Comment:
This is a bit heavy handed -- should we just be toggling the visibility? Or do we truly want to remove any potential styles?

Reply:
The goal here is to remove the js-set visibility and to have it move back up the cascade to have the visibility set by the CSS. I didn't see how to remove the specific `visibility` value set by the js above so I removed the attribute from the html element. I might be missing something though.